nilearn/regions/rena_clustering.py

```diff
@@ -171,11 +291,20 @@ def _weighted_connectivity_graph(X, mask_img):
     """
     n_features = X.shape[1]
 
-    edges, weight = _make_edges_and_weights(X, mask_img)
+    if isinstance(mask_img, SurfaceImage):
```

Comment:
I did not finish this section under the conditional for surface. It probably needs to be updated according to this comment https://github.com/nilearn/nilearn/pull/4205#discussion_r1461818983

And then the whole pipeline needs to be validated